### PR TITLE
Minor fixes from Coverity issues

### PR DESCRIPTION
**Pull request by pwithnall**
A smorgasbord of small, but sometimes significant, fixes.

### CODE REVIEW

**Comment by cgwalters** [comment]
The last two commits are dups of https://github.com/ostreedev/ostree/pull/1041

File: src/libostree/ostree-repo-commit.c [hunk 1]
```diff
@@ -486,12 +486,7 @@ create_regular_tmpfile_linkable_with_content (OstreeRepo *self,
           if (bytes_read < 0)
             return FALSE;
           else if (bytes_read == 0)
-            {
-              if (G_UNLIKELY (remaining > 0))
-                return glnx_throw (error, "Unexpected EOF with %" G_GUINT64_FORMAT "/%" G_GUINT64_FORMAT " bytes remaining", remaining, length);
-              else
-                break;
-            }
+            return glnx_throw (error, "Unexpected EOF with %" G_GUINT64_FORMAT "/%" G_GUINT64_FORMAT " bytes remaining", remaining, length);
```
Comment (cgwalters) [inline src/libostree/ostree-repo-commit.c]: This was also fixed in https://github.com/ostreedev/ostree/pull/1037

Comment (pwithnall) [inline src/libostree/ostree-repo-commit.c]: Gah, I lose the race.

**Comment by pwithnall** [comment]
I’ll wait for those PRs to go in, then rebase.

File: src/libostree/ostree-repo-commit.c (hunk 1 above)
Comment (pwithnall) [inline src/libostree/ostree-repo-commit.c]: Actually, no, the code pushed in #1037 is wrong. It still redundantly checks `(remaining > 0)` when that condition is guaranteed by the `while` condition.

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ a88bd4d

**Pull request closed without merging**